packages/components/src/DataList/components/DataListLayoutInternal/DataListItemInternal.tsx

```diff
@@ -0,0 +1,39 @@
+import classNames from "classnames";
+import React from "react";
+import { Checkbox } from "../../../Checkbox";
+import { useDataListContext } from "../../context/DataListContext";
+import styles from "../../DataList.css";
+import { DataListObject } from "../../DataList.types";
+
+export function ListItemInternal<T extends DataListObject>({
+  children,
+  item,
+}: {
+  children: JSX.Element;
+  item: T;
+}) {
+  const { selected, onSelect } = useDataListContext();
+
+  if (typeof selected !== "undefined" && onSelect) {
+    return (
+      <div
+        className={classNames(styles.selectable, {
+          [styles.selected]: selected?.length,
+        })}
+      >
+        <Checkbox
+          checked={selected?.includes(item.id)}
+          onChange={() => {
+            if (selected?.includes(item.id)) {
+              onSelect?.(selected?.filter(id => id !== item.id));
+            } else {
+              onSelect?.([...selected, item.id]);
+            }
+          }}
```

Comment:
do you mind on putting this under a `handleChange` function down below? 

Mostly to not muddy up this JSX. Better clean it now than it growing to an insane amount because no one wants to do it anymore in fear of "breaking" it